File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-1.cpp
class Solution {
public:
    int largestRectangleArea(vector<int> &heights)
    {
        int maxArea = heights[0];
        for (int i = 0; i < heights.size(); i++)
        {
            int l = i-1;
            int r = i+1;
            while (l >= 0 && heights[l] >= heights[i])
            {
                l--;
            }
            while (r < heights.size() && heights[r] >= heights[i])
            {
                r++;
            }
            cout << r << ",";
            cout << l << "\n";
            maxArea = max(maxArea, heights[i] * (r - l-1));
        }
        return maxArea;
    }

};

File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0 , len(heights)-1
        max_area = 0
        while l < r:
            h = min(heights[l],heights[r])
            area = h * (r-l)
            if area > max_area:
                max_area = area
            if h == heights[l]:
                l += 1
            else:
                r -= 1
        return max_area
